src/stk/molecular/atoms/atom.py

```diff
@@ -15,103 +17,107 @@ class Atom:
     Atoms of a particular element can be made with this
     class or with the subclass representing that element.
 
-    Examples
-    --------
-    *Initialization.*
+    Examples:
+
+        *Initialization.*
 
-    Initialization of an :class:`.Atom` can happen in one of two ways.
-    The atom can be initialized through the :class:`.Atom` class or
-    through the class representing the element.
+        Initialization of an :class:`.Atom` can happen in one of two
+        ways. The atom can be initialized through the :class:`.Atom`
+        class or through the class representing the element.
 
-    .. testcode:: initialization
+        .. testcode:: initialization
 
-        import stk
+            import stk
 
-        # h0 is an instance of the H class.
-        h0 = stk.Atom(id=0, atomic_number=1)
+            # h0 is an instance of the H class.
+            h0 = stk.Atom(id=0, atomic_number=1)
 
-        # h1 is also an instance of the H class.
-        h1 = stk.H(id=1)
+            # h1 is also an instance of the H class.
+            h1 = stk.H(id=1)
 
-    When the class corresponding to the element is used directly, the
-    ``atomic_number`` is not provided. Here are a few more examples.
+        When the class corresponding to the element is used directly,
+        the ``atomic_number`` is not provided. Here are a few more
+        examples.
 
-    .. testcode:: initialization
+        .. testcode:: initialization
 
-        # Both he0 and he1 are instances of the He class.
-        he0 = stk.Atom(id=2, atomic_number=2)
-        he1 = stk.He(id=3)
+            # Both he0 and he1 are instances of the He class.
+            he0 = stk.Atom(id=2, atomic_number=2)
+            he1 = stk.He(id=3)
 
-        # Both c0 and c1 are instances of the
-        # C class.
-        c0 = stk.Atom(id=4, atomic_number=6)
-        c1 = stk.C(id=5)
+            # Both c0 and c1 are instances of the
+            # C class.
+            c0 = stk.Atom(id=4, atomic_number=6)
+            c1 = stk.C(id=5)
 
     """
 
     # Maps each atomic number (int) to the relevant Atom subclass.
-    _elements = {}
+    _elements: Dict[int, Type['Atom']] = {}
+    _atomic_number: ClassVar[int]
 
-    def __init_subclass__(cls, **kwargs):
+    def __init_subclass__(cls: Type['Atom'], **kwargs) -> None:
         # Replace the default __init__() method of the subclass with
         # _subclass_init(). This is because the default __init__()
         # method takes an atomic_number parameter, but
         # _subclass_init() does not.
-        cls.__init__ = cls._subclass_init
+        cls.__init__ = cls._subclass_init  # type: ignore
```

Comment:
Why is this type: ignore?

Reply:
because I'm chaning the signature of the function, and therefore it's type. Mypy, quite rightly, complains about a type mismatch. However, in this case I think it's fine. The alternative would be to specify the innit function for each element, which is a lot of duplication, or add a new "implmentation" atom class, which the  element classes inherit, which has the init method defined. that would work pretty well, but i wanted to keep the inheritence hierarchy shallow.

Reply:
Cool cool, that makes sense. So this is a note for mypy basically?

Reply:
yeah. to be honest though. thinking about it now, i think the current way of replacing the init method is more complex and less obvious than just using simple inheritance. So i think I will just make the AtomImpl class.